src/git-operations.test.ts

```diff
@@ -1,67 +1,68 @@
 import execa from 'execa';
-import { didPackageChange } from './git-operations';
+import { didPackageChange, initializeGit, getTags } from './git-operations';
 
 // We don't actually use it, so it doesn't matter what it is.
 process.env.GITHUB_WORKSPACE = 'root';
 
 jest.mock('execa');
+const execaMock: jest.Mock<any, any> = execa as any;
+
+enum VERSIONS {
+  A = '1.0.0',
+  B = '1.0.1',
+  C = '1.1.0',
+}
+
+enum TAGS {
+  A = 'v1.0.0',
+  B = 'v1.0.1',
+  C = 'v1.1.0',
+}
+
+const PACKAGES: Readonly<Record<string, { name: string; dir: string }>> = {
+  A: { name: 'fooName', dir: 'foo' },
+  B: { name: 'barName', dir: 'bar' },
+};
+
+const RAW_MOCK_TAGS = `${Object.values(TAGS).join('\n')}\n`;
+
+const RAW_DIFFS: Readonly<Record<TAGS, string>> = {
+  [TAGS.A]: `packages/${PACKAGES.A.dir}/file.txt\npackages/${PACKAGES.B.dir}/file.txt\n`,
+  [TAGS.B]: `packages/${PACKAGES.A.dir}/file.txt\n`,
+  [TAGS.C]: `packages/${PACKAGES.B.dir}/file.txt\n`,
+};
 
 /**
- * ATTN: These cases are order-dependent.
- *
- * The first call to didPackageChange performs some initialization work.
- * Calls for valid tags, computed from packageData.manifest.version, are cached,
- * and future calls for the same tag perform no git operations.
+ * ATTN: This test suite is order-dependent due to git tag results being cached
```

Comment:
:grimacing: That is unfortunate. This seems to be caused by `TAGS` being a module-level variable. This would be much easier to test if it was an instance variable, passed in as a variable. Maybe out of scope for this PR though.